Src/HALALMock/Services/Time/Time.cpp

```diff
@@ -1,445 +1,168 @@
-uint64_t Time::mid_precision_tick = 0;
-
-void Time::init_timer(TIM_TypeDef* tim, TIM_HandleTypeDef* htim,
-                      uint32_t prescaler, uint32_t period,
-                      IRQn_Type interrupt_channel) {
-    TIM_ClockConfigTypeDef sClockSourceConfig = {0};
-    TIM_MasterConfigTypeDef sMasterConfig = {0};
-
-    htim->Instance = tim;
-    htim->Init.Prescaler = prescaler;
-    htim->Init.CounterMode = TIM_COUNTERMODE_UP;
-    htim->Init.Period = period;
-    htim->Init.ClockDivision = TIM_CLOCKDIVISION_DIV1;
-    htim->Init.AutoReloadPreload = TIM_AUTORELOAD_PRELOAD_DISABLE;
-    HAL_TIM_Base_Init(htim);
-
-    sClockSourceConfig.ClockSource = TIM_CLOCKSOURCE_INTERNAL;
-    HAL_TIM_ConfigClockSource(htim, &sClockSourceConfig);
-
-    sMasterConfig.MasterOutputTrigger = TIM_TRGO_RESET;
-    sMasterConfig.MasterSlaveMode = TIM_MASTERSLAVEMODE_DISABLE;
-
-    HAL_TIMEx_MasterConfigSynchronization(htim, &sMasterConfig);
-
-    HAL_NVIC_SetPriority(interrupt_channel, 0, 0);
-    HAL_NVIC_EnableIRQ(interrupt_channel);
-}
-
-void Time::start() {
-    __HAL_RCC_TIM7_CLK_ENABLE();
-    __HAL_RCC_TIM24_CLK_ENABLE();
-    __HAL_RCC_TIM23_CLK_ENABLE();
-    Time::init_timer(TIM2, &htim2, 0, HIGH_PRECISION_MAX_ARR, TIM2_IRQn);
-    Time::init_timer(TIM5, &htim5, 0, HIGH_PRECISION_MAX_ARR, TIM5_IRQn);
-    Time::init_timer(TIM24, &htim24, 0, HIGH_PRECISION_MAX_ARR, TIM24_IRQn);
-    Time::init_timer(TIM7, &htim7, 275, 1000, TIM7_IRQn);
-    HAL_TIM_Base_Start_IT(low_precision_timer);
-
-    if (global_timer != nullptr) {
-        if (timer32_by_timer32handler_map.find(global_timer) ==
-            timer32_by_timer32handler_map.end()) {
-            ErrorHandler("Global Timer pointer is not valid");
-            global_timer = nullptr;
-        } else {
-            hal_enable_timer(global_timer);
-            Time::init_timer(
-                timer32_by_timer32handler_map[global_timer], global_timer, 0,
-                HIGH_PRECISION_MAX_ARR,
-                timer32interrupt_by_timer32handler_map[global_timer]);
-            HAL_TIM_Base_Start_IT(global_timer);
-        }
+void Time::start(double simulation_speed) {
+    std::lock_guard<std::mutex> lock(mutex);
+    if (running) {
+        return; // Already running
     }
-    for (auto timer : high_precision_timers) {
-        if (timer32_by_timer32handler_map.find(timer) ==
-            timer32_by_timer32handler_map.end()) {
-            ErrorHandler("Global Timer pointer is not valid");
-            high_precision_timers.erase(timer);
-        } else {
-            hal_enable_timer(timer);
-            Time::init_timer(timer32_by_timer32handler_map[timer], timer, 0,
-                             HIGH_PRECISION_MAX_ARR,
-                             timer32interrupt_by_timer32handler_map[timer]);
-            Time::available_high_precision_timers.push(timer);
-        }
+    if (simulation_speed <= 0.0) {
+        //TODO: ErrorHandler("Simulation speed must be positive");
```

Comment:
Would already add an error handler here with the TODO to maybe change it in the future for the simulator-specific logging system we use